crates/uv/src/settings.rs

```diff
@@ -2452,30 +2457,33 @@ impl ResolverSettings {
 
 impl From<ResolverOptions> for ResolverSettings {
     fn from(value: ResolverOptions) -> Self {
+        let index_locations = IndexLocations::new(
+            value
+                .index
+                .into_iter()
+                .flatten()
+                .chain(value.extra_index_url.into_iter().flatten().map(Index::from))
+                .chain(value.index_url.into_iter().map(Index::from))
+                .collect(),
+            value
+                .find_links
+                .into_iter()
+                .flatten()
+                .map(Index::from)
+                .collect(),
+            value.no_index.unwrap_or_default(),
+        );
+        let url_auth_policies = (&index_locations).into();
         Self {
-            index_locations: IndexLocations::new(
-                value
-                    .index
-                    .into_iter()
-                    .flatten()
-                    .chain(value.extra_index_url.into_iter().flatten().map(Index::from))
-                    .chain(value.index_url.into_iter().map(Index::from))
-                    .collect(),
-                value
-                    .find_links
-                    .into_iter()
-                    .flatten()
-                    .map(Index::from)
-                    .collect(),
-                value.no_index.unwrap_or_default(),
-            ),
+            index_locations,
             resolution: value.resolution.unwrap_or_default(),
             prerelease: value.prerelease.unwrap_or_default(),
             fork_strategy: value.fork_strategy.unwrap_or_default(),
             dependency_metadata: DependencyMetadata::from_entries(
                 value.dependency_metadata.into_iter().flatten(),
             ),
             index_strategy: value.index_strategy.unwrap_or_default(),
+            url_auth_policies,
```

Comment:
Why store this on the settings type at all, rather than create it from `index_locations` wherever we initialize the network client?

Reply:
This was just leftover from different approaches to threading this through. I've updated it.